tests/ui/const-generics/occurs-check/unify-fixpoint.rs

```diff
@@ -1,3 +1,8 @@
+// -Zunstable-options added as test for ICE #97725 (left == right)`
```

Comment:
How does `-Zunstable-options` change behavior of this test?

Reply:
I assume the thing that causes that test to fail is not `-Znunstable-options`, but just debug assertions. We don't test those in CI afaict.

Reply:
It would have caught the original ice which is reason enough to add it imo.

I do run x.py test with debug assertions from time to time.

Reply:
No, I am asking why you believe `-Zunstable-options` would have caught the original ICE. `-Zunstable-options` afaict does nothing on its own -- it's required to pass *other* unstable flags, but it doesn't do anything behavioral on its own.

Reply:
I have been reporting several ices where `-Zunstable-options` was crucial to reproducing the ICE, with no other flags being involved.
https://github.com/rust-lang/rust/issues?q=is%3Aissue+%22-Zunstable-options%22++author%3Amatthiaskrgr+
As to why that happens, no idea :shrug: